plugin.video.montreal.greek-tv/resources/lib/directory.py

```diff
@@ -0,0 +1,230 @@
+
+            try:
+                url += '&url=%s' % quote_plus(i['url'])
+            except BaseException:
+                pass
+            try:
+                url += '&title=%s' % quote_plus(i['title'])
+            except KeyError:
+                try:
+                    url += '&title=%s' % quote_plus(i['title'].encode('utf-8'))
+                except KeyError:
+                    pass
+            except BaseException:
+                pass
+
+            try:
+                url += '&image=%s' % quote_plus(i['image'])
+            except KeyError:
+                try:
+                    url += '&image=%s' % quote_plus(i['image'].encode('utf-8'))
+                except KeyError:
+                    pass
+            except BaseException:
+                pass
+            try:
+                url += '&name=%s' % quote_plus(i['name'])
+            except KeyError:
+                try:
+                    url += '&name=%s' % quote_plus(i['name'].encode('utf-8'))
+                except KeyError:
+                    pass
+            except BaseException:
+                pass
+            try:
+                url += '&year=%s' % quote_plus(i['year'])
+            except BaseException:
+                pass
+            try:
+                url += '&plot=%s' % quote_plus(i['plot'])
+            except KeyError:
+                try:
+                    url += '&plot=%s' % quote_plus(i['plot'].encode('utf-8'))
+                except KeyError:
+                    pass
+            except BaseException:
+                pass
+            try:
+                url += '&genre=%s' % quote_plus(i['genre'])
+            except KeyError:
+                try:
+                    url += '&genre=%s' % quote_plus(i['genre'].encode('utf-8'))
+                except KeyError:
+                    pass
+            except BaseException:
+                pass
+            try:
+                url += '&dash=%s' % quote_plus(i['dash'])
+            except BaseException:
+                pass
+            try:
+                url += '&query=%s' % quote_plus(i['query'])
+            except BaseException:
+                pass
+
+            cm = []
+            menus = i['cm'] if 'cm' in i else []
+
+            for menu in menus:
+                try:
+                    try:
+                        tmenu = control.lang(menu['title']).encode('utf-8')
+                    except BaseException:
+                        tmenu = menu['title']
+                    qmenu = urlencode(menu['query'])
+                    cm.append((tmenu, 'RunPlugin(%s?%s)' % (sysaddon, qmenu)))
+                except BaseException:
+                    pass
+
+            meta = dict((k, v) for k, v in iteritems(i) if not k == 'cm' and not v == '0')
+
+            if mediatype is not None:
+                meta['mediatype'] = mediatype
+
+            item = control.item(label=label, iconImage=image, thumbnailImage=image)
```

Comment:
iconImage and thumbnail image are both deprecated please remove and use setArt on the listitem only

Reply:
ok.